src/main/kotlin/JobState.kt

```diff
@@ -189,6 +205,34 @@ class JobState<U: Any> constructor(val value: U?, val interceptors: List<WorkerI
         return process(lastInterceptor, JobStatus.TransientFailure)
     }
 
+    private suspend fun <R: Any> processEither(fn: suspend(U) -> Either<R>, msg: String? = null): JobState<R> {
+        var eitherValue: Either<R>? = null
+        val lastInterceptor = WorkerInterceptor {_, _ ->
+            if(inProgress()) {
+                try{
```

Comment:
For clarity, avoiding mutable vars, and for avoiding catching exceptions caused by Franz's code I think the approach is more appropriate. 
```
val eitherValue = try {
  fn(this.value!!)
} catch (e: Throwable) {
// Catch stuff here. [...]
}
// Logic handling eitherValue here.. [...]